internal/net/net.go

```diff
@@ -29,12 +29,12 @@ func IsIPV6(ip _net.IP) bool {
 
 // IsPortAvailable checks if a TCP port is available or not
 func IsPortAvailable(p int) bool {
-	conn, err := _net.Dial("tcp", fmt.Sprintf(":%v", p))
+	ln, err := _net.Listen("tcp", fmt.Sprintf(":%v", p))
```

Comment:
Cant it be a problem if we try to bind nginx to a specific IP, but here you check all the interfaces?

Reply:
https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#bind-address <- we should probably check against this address specifically:

1) If it does not exists, we may try binding to every address
2) If this exists, then we should try to bind specifically to this address

Reply:
Yes, you are right, I just did a workaround let it can run in my cluster. I will try to change it under your suggested.

Reply:
@rikatz but the function was use in init check, so it will change so many things.

Reply:
hum indeed, this value is in ConfigMap and it's not a flag...

OK, so overall this problem would still happen if we have something listening in the port, I think we can go forward with this one.